sphericart-torch/CMakeLists.txt

```diff
@@ -112,18 +112,25 @@ find_package(Torch 1.13 REQUIRED)
 file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/_build_torch_version.py "BUILD_TORCH_VERSION = '${Torch_VERSION}'")
 
 add_library(sphericart_torch SHARED
+    "include/sphericart/torch_cuda_wrapper.hpp"
     "include/sphericart/torch.hpp"
     "include/sphericart/autograd.hpp"
     "src/autograd.cpp"
     "src/torch.cpp"
 )
 
 if (CMAKE_CUDA_COMPILER)
-    target_sources(sphericart_torch PUBLIC "src/cuda.cu")
+    target_sources(sphericart_torch PUBLIC "src/cuda_base.cu")
 else()
     target_sources(sphericart_torch PUBLIC "src/cuda_stub.cpp")
 endif()
 
+if (CMAKE_CUDA_COMPILER)
+    target_sources(sphericart_torch PUBLIC "src/torch_cuda_wrapper.cpp")
+else()
+    target_sources(sphericart_torch PUBLIC "src/torch_cuda_wrapper_stub.cpp")
+endif()
+
```

Comment:
I would merge the two if here!

Reply:
I think it's easier for me to provide a CUDA C example of how to use these functions when we've completed relocating the CUDA code into `sphericart`.

Traditionally CUDA C is function-based. If you want to make an additional wrapper around these we'd need to write it...

Reply:
A C example is fine with me as well!